src/com/etendoerp/dependencymanager/datasource/AddSubDependencyDS.java

```diff
@@ -0,0 +1,296 @@
+    return getData(parameters, 0, Integer.MAX_VALUE).size();
+  }
+
+  /**
+   * Retrieves a list of data maps based on the provided parameters and pagination settings.
+   *
+   * <p>This method fetches a subset of the data starting from {@code startRow} to {@code endRow}.</p>
+   *
+   * @param parameters
+   *     The map of parameters used to filter or specify the data.
+   * @param startRow
+   *     The starting row index for pagination.
+   * @param endRow
+   *     The ending row index for pagination.
+   * @return A list of data maps containing the retrieved data.
+   * @throws OBException
+   *     If an error occurs during data retrieval.
+   */
+  @Override
+  protected List<Map<String, Object>> getData(Map<String, String> parameters, int startRow, int endRow) {
+    List<Map<String, Object>> result;
+    try {
+      final String strETDEPPackageVersionId = parameters.get("@ETDEP_Package_Version.id@");
+      final PackageVersion packageVersion = OBDal.getInstance().get(PackageVersion.class, strETDEPPackageVersionId);
+
+      result = getGridData(parameters, packageVersion);
+
+    } catch (Exception e) {
+      throw new OBException(e.getMessage());
+    }
+
+    return result;
+  }
+
+  /**
+   * Retrieves the grid data based on the package version and parameters.
+   *
+   * <p>This method processes the package dependencies, builds a map of the dependencies,
+   * and returns a filtered and sorted list of data maps.</p>
+   *
+   * @param parameters
+   *     The map of parameters used to filter the data.
+   * @param packageVersion
+   *     The version of the package to retrieve the dependencies for.
+   * @return A list of data maps containing the dependency information.
+   * @throws JSONException
+   *     If there is an error processing the JSON data.
+   */
+  private List<Map<String, Object>> getGridData(Map<String, String> parameters,
+      PackageVersion packageVersion) throws JSONException {
+
+    List<Map<String, Object>> result = new ArrayList<>();
+
+    List<PackageDependency> dependenciesList = packageVersion.getETDEPPackageDependencyList();
+    removeDependecyCore(dependenciesList);
+
+    Map<String, PackageDependency> dependencyMap = new HashMap<>();
+    Map<String, String> parentMap = new HashMap<>();
+
+    DependencyTreeBuilder.addDependenciesWithParents(dependenciesList, dependencyMap, parentMap);
+
+    List<PackageDependency> dependencyList = new ArrayList<>(dependencyMap.values());
+    for (PackageDependency dependency : dependencyList) {
+      Map<String, Object> map = new HashMap<>();
+      map.put(DependencyManagerConstants.GROUP, dependency.getGroup());
+      map.put(DependencyManagerConstants.ARTIFACT, dependency.getArtifact());
+      map.put(DependencyManagerConstants.VERSION, dependency.getVersion());
+      map.put(DependencyManagerConstants.ID, dependency.getId());
+
+      String parentArtifact = parentMap.get(dependency.getId());
+      if (parentArtifact != null) {
```

Comment:
Should this also have an isEmpty check? Is it necessary to do so?

Reply:
No :)